lms/djangoapps/oauth_dispatch/dot_overrides.py

```diff
@@ -0,0 +1,41 @@
+"""
+Classes that override default django-oauth-toolkit behavior
+"""
+
+from django.contrib.auth import authenticate as django_authenticate, get_user_model
+from oauth2_provider.oauth2_validators import OAuth2Validator
+
+
+def authenticate(username, password):
```

Comment:
I think my idea was to make a function with the same interface as [`django.contrib.auth.authenticate`](https://github.com/django/django/blob/master/django/contrib/auth/__init__.py#L62-85), so we could use it as a drop-in replacement.  On further consideration, I'm not sure that's worthwhile, but we might consider a custom `AUTHENTICATION_BACKEND`, which might circumvent having custom logic all over our app.  On the other hand ensuring that doesn't cause regressions sounds big and scary, so maybe we shouldn't do that in the context of this bugfix.